src/index.js

```diff
@@ -151,6 +144,31 @@ export default class ScrollBehavior {
     });
   }
 
+  _setScrollRestoration = () => {
+    if (this._hasSetScrollRestoration) {
+      // It's possible that the `pageshow` event occurs after we initialise
+      return;
+    }
+    if (
+      'scrollRestoration' in window.history &&
+      // Unfortunately, Safari on iOS freezes for 2-6s after the user swipes to
+      // navigate through history with scrollRestoration being 'manual', so we
+      // need to detect this browser and exclude it from the following code
+      // until this bug is fixed by Apple.
+      !isMobileSafari()
+    ) {
+      this._oldScrollRestoration = window.history.scrollRestoration;
+      try {
+        window.history.scrollRestoration = 'manual';
+      } catch (e) {
+        this._oldScrollRestoration = null;
+      }
+    } else {
+      this._oldScrollRestoration = null;
+    }
+    this._hasSetScrollRestoration = true;
+  };
+
   _restoreScrollRestoration = () => {
     /* istanbul ignore if: not supported by any browsers on Travis */
     if (this._oldScrollRestoration) {
```

Comment:
so just bail if `this._hasSetScrollRestoration` is `false`?

Reply:
Ive replaced this with just `_oldScrollRestoration` as you suggested - does that look good?